drop python 2 from entire sympy

sidhu1012 opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->
fixes #18816 

#### Brief description of what is fixed or changed
Dropped python 2 from sympy

#### Other comments


#### Release Notes

<!-- Write the release notes for this release below. See
https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more information
on how to write release notes. The bot will check your release notes
automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

sidhu1012 commented:
Why did this test fail in Travis?


Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.8.0/lib/python3.8/site-packages/sympy-1.7.dev0-py3.8.egg/sympy/testing/runtests.py", line 1340, in _timeout
    function()
  File "/home/travis/virtualenv/python3.8.0/lib/python3.8/site-packages/sympy-1.7.dev0-py3.8.egg/sympy/testing/pytest.py", line 175, in func_wrapper
    func()
  File "/home/travis/virtualenv/python3.8.0/lib/python3.8/site-packages/sympy-1.7.dev0-py3.8.egg/sympy/assumptions/tests/test_query.py", line 2105, in test_known_facts_consistent
    assert f.read() == \
AssertionError

oscarbenjamin commented:
I'm not sure but maybe just revert the changes in sympy/assumptions

sidhu1012 commented:
Why are these files conflicting

sidhu1012 commented:
@oscarbenjamin I think it can be merged now

oscarbenjamin commented on sympy/parsing/autolev/_antlr/autolevlexer.py:
```diff
@@ -5,193 +5,192 @@
 # Generated with antlr4
 #    antlr4 is licensed under the BSD-3-Clause License
 #    https://github.com/antlr/antlr4/blob/master/LICENSE.txt
-from __future__ import print_function
```
This is an auto-generated file so we shouldn't change it in this PR

oscarbenjamin commented on sympy/parsing/autolev/_antlr/autolevparser.py:
```diff
@@ -5,230 +5,229 @@
 # Generated with antlr4
 #    antlr4 is licensed under the BSD-3-Clause License
 #    https://github.com/antlr/antlr4/blob/master/LICENSE.txt
-from __future__ import print_function
```
Also autogenerated

oscarbenjamin commented on sympy/parsing/latex/_antlr/latexlexer.py:
```diff
@@ -1,4 +1,5 @@
+
 # encoding: utf-8
 
 # *** GENERATED BY `setup.py antlr`, DO NOT EDIT BY HAND ***
```
Also autogenerated

oscarbenjamin commented:
We shouldn't change automatically generated files in a change like this. The setucp.cfg lists these files
```
exclude = sympy/parsing/latex/_antlr/*,
          sympy/parsing/autolev/_antlr/*,
          sympy/parsing/autolev/test-examples/*,
          sympy/integrals/rubi/*
```
but there could be others.

sidhu1012 commented:
@oscarbenjamin  Is it good to go now?

oscarbenjamin commented:
It looks like the autolev automatically generated files are still changed

sidhu1012 commented:
> It looks like the autolev automatically generated files are still changed

No they aren't changes. Yes it's showing they are changes but have a look at the changes , there isn't any.

sidhu1012 commented:
@oscarbenjamin Parsing module was updates which cause conflict, so I resolved that conflict , the changes it's showing are irrelevant as the modifications it is showing are same as the lines being removed. I dont know why it is showing such behaviour but this PR is no longer modifying auto generated files

oscarbenjamin commented:
> Yes it's showing they are changes but have a look at the changes , there isn't any.

There are changes to the autolev files.

sidhu1012 committed:
Remove unwanted python2 code, command used- pyup_dirs --py3-plus
--recursive --keep-percent-format sympy, removed all py2 code instances
except code tokens and percent formatting from all files except auto-generated files

smitgajjar commented on sympy/parsing/ast_parser.py:
```diff
@@ -19,7 +19,6 @@
 """
 
-from __future__ import print_function, division
 
```
Remove this redundant newline after import

smitgajjar commented on sympy/parsing/sympy_parser.py:
```diff
@@ -1,6 +1,5 @@
 """Transform a string with Python-like source code into SymPy expression. """
 
-from __future__ import print_function, division
 
```
Here too!

smitgajjar commented on sympy/physics/continuum_mechanics/beam.py:
```diff
@@ -3,7 +3,6 @@
 """
 
-from __future__ import print_function, division
 
```
Here and many other places

sidhu1012 replied:
The line from which imports are removed are also removed automatically.

smitgajjar replied:
Next line!

sidhu1012 replied:
There are 214 files that are changed and is done by tool , so I dont think it's necessary to remove each line by hand.

oscarbenjamin replied:
The fact that this is done automatically does not mean that it isn't necessary to check manually as well afterwards. @sidhu1012 have you been through the diff for this PR checking whether the line spacing is consistent with the linespacing in these files before applying the script?

sidhu1012 replied:
Yes

oscarbenjamin replied:
The examples highlighted by @smitgajjar seem to be cases where the new linespacing is not consistent with how it was before.

asmeurer commented:
If the changes are generated automatically, you should note what command was used in the commit message.

sidhu1012 commented:
> If the changes are generated automatically, you should note what command was used in the commit message.

Done

sidhu1012 commented:
Please Review. I think it's good to go now.

sidhu1012 commented:
@oscarbenjamin @asmeurer @smitgajjar I think it can be merged now as I have cross checked it doesn't make any changes to autogenerated files and I have also corrected the line spacing. It's consistent now as it was with python 2 imports.

sidhu1012 commented:
Can someone please review

sidhu1012 commented:
Someone please review, I think it's ready to merge

sidhu1012 commented:
Since no one is reviewing, @oscarbenjamin  please review it when free.

sidhu1012 commented:
Can someone please review this  PR

sidhu1012 commented:
Review please

oscarbenjamin commented:
This looks good. Can you squash the commits into a single commit and give it a commit message that clearly says what command was used and what effect it has? Something like:
```
maint: remove no longer needed python 2 code

This commit was created with
  $ <command used>
This removes all examples of
1. Explicit subclassing of object (i.e. class X(object):). In Python 3 it is not necessary to subclass object explicitly as old-style classes are removed.
2. Unnecessary brackets e.g. ((x+y))
...<more explanation of changes>
```

sidhu1012 commented:
Yes sure

sidhu1012 commented:
@oscarbenjamin Done!!

sidhu1012 commented:
@oscarbenjamin Please review, hope it's good to go

oscarbenjamin commented:
Looks good. Thanks!
